src/compiler/checker.ts

```diff
@@ -2851,18 +2851,18 @@ namespace ts {
 
         function reportInvalidImportEqualsExportMember(node: ImportDeclaration | ExportDeclaration | VariableDeclaration, name: Identifier, declarationName: string, moduleName: string) {
             if (moduleKind >= ModuleKind.ES2015) {
-                const message = compilerOptions.esModuleInterop ? Diagnostics._0_can_only_be_imported_by_using_a_default_import :
+                const message = getESModuleInterop(compilerOptions) ? Diagnostics._0_can_only_be_imported_by_using_a_default_import :

```

Comment:
Should we have a normalized version of CompilerOptions so that we don't have to call those function again and again?

Reply:
Eh, it might be nice, but then we'd have to track it through the typesystem and delicate where we have processed vs unprocessed compiler options.